cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/batch/mapreduce/MapReduceSinkContext.java

```diff
@@ -53,10 +56,11 @@ public void addOutput(final String datasetName) {
 
   @Override
   public void addOutput(final String datasetName, final Map<String, String> arguments) {
+    final Output actualOutput = getOutput(Output.ofDataset(datasetName, arguments));
```

Comment:
I think they are not needed as for both mapreduce, the arguments will be used to construct `DatasetOutputFormatProvider`, which now be replaced to `NullOutputFormatProvider`. For spark, the arguments is needed if we want to write to a dataset, which is not we want in preview